pilot/pkg/serviceregistry/kube/controller/network.go

```diff
@@ -313,6 +330,79 @@ func (n *networkManager) getGatewayDetails(svc *model.Service) []model.NetworkGa
 	return nil
 }
 
+func (n *networkManager) watchGatewayResources(c *Controller, stop <-chan struct{}) {
+	n.gatewayResourceClient = kclient.NewDelayedInformer(c.client, gvr.KubernetesGateway, kubetypes.StandardInformer, kubetypes.Filter{
+		LabelSelector: label.TopologyNetwork.Name,
+		FieldSelector: "spec.gatewayClassName=istio",
+	})
+	registerHandlers[controllers.Object](c, n.gatewayResourceClient, "Gateways", n.handleGatewayResource, nil)
+	go n.gatewayResourceClient.Start(stop)
+}
+
+// handleGateway resource adds a NetworkGateway for each combination of address and auto-passthrough listener
+// discovering duplicates from the generated Service is not a huge concern as we de-duplicate in NetworkGateways
+// which returns a set, although it's not totally efficient.
+func (n *networkManager) handleGatewayResource(_ controllers.Object, obj controllers.Object, event model.Event) error {
+	gw, ok := obj.(*v1beta1.Gateway)
+	if !ok {
+		return nil
+	}
+
+	gatewaysChanged := false
+	n.Lock()
+	defer func() {
+		n.Unlock()
+		if gatewaysChanged {
+			n.NotifyGatewayHandlers()
+		}
+	}()
+
+	previousGateways := n.gatewaysFromResource[gw.UID]
+
+	if event == model.EventDelete {
+		gatewaysChanged = len(previousGateways) > 0
+		delete(n.gatewaysFromResource, gw.UID)
+		return nil
+	}
+
+	autoPassthrough := func(l v1beta1.Listener) bool {
```

Comment:
Can you add a comment somewhere on where is this used ? Is it to configure the istio gateway to enable auto-passthrough in the backend ? Or to detect if the remote gateway has the 15443 port and can be used to route ? 

If the later - I don't think there is a strict requirement that any E/W gateway is forwarding EVERYTHING. Even in current implementation an admin may use NetworkPolicy and other controls to limit the pods that are publicly exposed ( because our E/W remains a pretty serious risk ). 
Having port 15443 with a certain name tells us this is a gateway that may forward mTLS to backends in a different network.